app/app.go

```diff
@@ -836,12 +853,20 @@ func (app *App) Name() string { return app.BaseApp.Name() }
 
 // BeginBlocker application updates every begin block
 func (app *App) BeginBlocker(ctx sdk.Context, req abci.RequestBeginBlock) abci.ResponseBeginBlock {
-	return app.mm.BeginBlock(ctx, req)
+	res := app.mm.BeginBlock(ctx, req)
+	for _, s := range app.streamingServices {
+		s.StreamBeginBlocker(ctx, req, res)
+	}
```

Comment:
Should this (and the similar new lines in `EndBlocker` be fired off as separate `go` routines? E.g.:
```golang
go func() {
	for _, s := range app.streamingServices {
		s.StreamBeginBlocker(ctx, req, res)
	}
}()
```

Pro: The `BeginBlocker` function doesn't have to wait on any of the calls to `StreamBeginBlocker`.
Con: Possible memory concerns about holding onto the `req` and `res` values outside the running of the `BeginBlocker` or `EndBlocker` functions. Also possible concurrency concerns if one isn't done yet, but `BeginBlocker` or `EndBlocker` is called again.

Reply:
The intent is to stay in sync with the external service. Otherwise, the node may jump a head blocks when the external service fails. Ideally, streaming to external service would run on a non-validator node. This way there's no concern of performance issues. We should actually document this in the config. We can put a `goroutine` in there but it will need to be with a `wg` (with Kafka we'll produce [asynchronously](https://github.com/confluentinc/confluent-kafka-go#examples)) .

I'm actually changing the design to only have one service enabled at a time. We should support multiple services. I don't know what I was thinking there? We only need to push out to one external service and have consumer applications `fan-out` from the external service.

```toml
[stremaing]

# trace, kafka, postgresql
service = ""
```